code/modules/spells/spell_types/shadow_walk.dm

```diff
@@ -46,11 +48,16 @@
 	STOP_PROCESSING(SSobj, src)
 	. = ..()
 
-/obj/effect/dummy/phased_mob/shadow/process()
+/obj/effect/dummy/phased_mob/shadow/process(delta_time)
+	var/turf/T = get_turf(src)
+	var/light_amount = T.get_lumcount()
 	if(!jaunter || jaunter.loc != src)
 		qdel(src)
+	if (light_amount < 0.2 && (!QDELETED(jaunter))) //heal in the dark
+		jaunter.heal_overall_damage((SHADOW_REGEN_RATE * delta_time), (SHADOW_REGEN_RATE * delta_time), 0, BODYPART_ORGANIC)
```

Comment:
delta_time is defined as wait * 0.1 for processing subsystems, since ssobject is on a 20ms wait, that means each process the nightmare heals 2 * SHADOW_REGEN_RATE, in this case 6 damage per second. This seems a bit high, maybe tone it down a tad? say set regen rate to 1.5/1?